src/hooks/useEffectAsync.ts

```diff
@@ -0,0 +1,37 @@
+import React, { useEffect } from 'react';
+
+type AsyncEffect = (signal: AbortSignal) => Promise<void | (() => void)> | void;
+
+/**
+ * Wrapper for useEffect which provides an abort signal that can be used for mounted state checks,
+ * as well as for aborting async https requests in progress if added to header.
+ *
+ * example:
+ * useEffectAsync(async (signal) => {
+ *       const response = await request();
+ *       if (!signal.aborted) {
+ *           setState("some value")
+ *       }
+ *
+ *       return () => {
+ *           optionalCleanup();
+ *       }
+ * }, [])
+ */
+export function useEffectAsync(effect: AsyncEffect, deps: React.DependencyList = []): void {
```

Comment:
Could you add unit test as well?
We use `@testing-library/react-hooks` for test hooks without components.
An example on that:
```ts
import { act, renderHook } from '@testing-library/react-hooks';
...
        it('should return with the instCode description when the plant exists', () => {
            // given
            const instCode = 'JCA';
            const { result } = renderHook(() => useInstCodeDescription(instCode));

            // when
            expect(result.current).toBe('mock JCA description');
        });

```
More info about usage can be found here:
https://www.npmjs.com/package/@testing-library/react-hooks

Reply:
As discussed: I'll add unit test for this later, just to not hold up the PR for long.

Reply:
Poor excuse for not writing tests, later === never in almost all cases

Reply:
Probably some context is missing here: I'm also have a task in echoUtils at the moment, where I need to add tests.
My plan was to add this test there.

But ofc you're right on being more strict about this: I'll add the test to this PR then.

Reply:
@hkot1991 added tests :)